lib/cos.js

```diff
@@ -0,0 +1,134 @@
+
+exports.initialize = function initializeDataSource(dataSource, callback) {
+	//console.log(dataSource.settings);
+	dataSource.connector = new COSConnector(dataSource.settings);
+	process.nextTick(function () {
+		callback && callback();
+	});
+};
+var Connector = require('loopback-connector').Connector;
+
+require('util').inherits(COSConnector, Connector);
+
+const IBM = require('ibm-cos-sdk');
+
+exports.COSConnector = COSConnector;
+
+function COSConnector(dataSourceProps) {
+
+	this.endpoint = dataSourceProps.endpoint;
+	this.apiKeyId = dataSourceProps.apiKeyId;
+	this.ibmAuthEndpoint = 'https://iam.ng.bluemix.net/oidc/token';
+	this.serviceInstanceId = dataSourceProps.serviceInstanceId;
+	this.bucket = dataSourceProps.bucket;
+	this.response = [];
+	this._models = {};
+
+	this.config = {
+		endpoint: this.endpoint,
+		apiKeyId: this.apiKeyId,
+		ibmAuthEndpoint: this.ibmAuthEndpoint,
+		serviceInstanceId: this.serviceInstanceId,
+	};
+	this.cos = new IBM.S3(this.config);
+
+}
+
+
+//creates entry in bucket
+COSConnector.prototype.create = function create(model, data, callback){
+
+	var key = guid();
+	var dataStr = JSON.stringify(data); 
+
+	return this.cos.putObject({
+		'Bucket': this.bucket, 
+		'Key': key, 
+		'Body': dataStr,
+	}).promise()
+		.then(() => {
+    
+			data.id=key;
+			this.response.push(data);
+			callback(null, data.id);
+		})
+		.catch((e) => (`ERROR1: ${e.code} - ${e.message}\n`));	
+};
+
+//deletes entry to bucket based on object name
+exports.COSConnector.prototype.destroyAll = function destroy(model,where,callback){
+
+	var dataStr = where.id;
```

Comment:
What if `where.id` is not set? I think you should detect that case and throw an error with a descriptive message that will allow the user to understand what went wrong.